test/BookingApi.UnitTests/Fixtures/FakeAPIClient.cs

```diff
@@ -0,0 +1,220 @@
+                message.Headers.TryAddWithoutValidation("Authorization", $"{_privateKey}:{authentication}");
+                message.Headers.Date = requestDateTime;
+                return message;
+            });
+
+            return await SendWithLock(httpRequest);
+        }
+
+        public async Task<IShipmentTrackingResponse> TrackShipment(Action<IShipmentTrackingRequest> requestBuilder)
+        {
+            _httpClient = SetupHttpClient();
+            if (string.IsNullOrEmpty(_secretKey) || string.IsNullOrEmpty(_privateKey))
+                throw new NotImplementedException();
+
+            var request = new TrackShipmentRequest(_httpClient.BaseAddress.ToString());
+            requestBuilder(request);
+
+            var rawJson = JsonConvert.SerializeObject(request, _serializerSettings);
+            var httpRequest = new HttpRequest<TrackShipmentRequest, TrackShipmentResponse>(request);
+
+            httpRequest.ConstructRequest(() => {
+                var requestDateTime = DateTime.Now;
+
+                var message = new HttpRequestMessage(request.Method, request.Endpoint) {
+                    Content = new StringContent(rawJson)
+                };
+
+                message.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json");
+
+                var authentication = SignRequest(request.Method, rawJson, message.Content.Headers.ContentType.ToString(),
+                    $"/api/shipment/{request.Barcode}", requestDateTime);
+
+                message.Headers.TryAddWithoutValidation("Authorization", $"{_privateKey}:{authentication}");
+                message.Headers.Date = requestDateTime;
+                return message;
+            });
+
+            return await SendWithLock(httpRequest);
+        }
+
+        public void UseLiveApi() => _isTestingMode = false;
+
+        public void UseStagingApi() => _isTestingMode = true;
+        private string SignRequest(HttpMethod method, string rawJson, string contentType, string endpoint, DateTime dateTime)
+        {
+            var md5 = MD5.Create();
+
+            var authBuilder = new StringBuilder();
+            authBuilder.Append(method).Append("\n");
+            authBuilder.Append(BitConverter.ToString(md5.ComputeHash(Encoding.UTF8.GetBytes(rawJson)))
+                .Replace("-", "").ToLowerInvariant()).Append("\n");
+            authBuilder.Append(contentType).Append("\n");
+            authBuilder.Append(dateTime.ToUniversalTime().ToString("r")).Append("\n");
+            authBuilder.Append(endpoint);
+
+            var hmacsha1 = new HMACSHA1 { Key = Encoding.UTF8.GetBytes(_secretKey) };
+            var signBytes = Encoding.UTF8.GetBytes(authBuilder.ToString());
+            var hashResult = hmacsha1.ComputeHash(signBytes);
+
+            return Convert.ToBase64String(hashResult);
+        }
+
+        private static async Task<TResponse> SendWithLock<TRequest, TResponse>(IHttpRequest<TRequest, ErrorResponse, TResponse> httpRequest)
```

Comment:
Probably don't need to lock when you are mocking a HttpClient result. I have it in the live one to stop multiple live requests from modifying the client beforehand